include/seqan/platform/platform_gcc.h

```diff
@@ -39,9 +39,20 @@
  * @signature #define PLATFORM_GCC
  */
 
+// GNU COMPATIBLE
 #ifndef PLATFORM_GCC
 #define PLATFORM_GCC
 
+#if !defined(__clang__) && !defined(__ICC) && !defined(PLATFORM_GNU)
+// ACTUALLY GNU
+#define PLATFORM_GNU
+#endif
+
+#if defined(__clang__) && !defined(__ICC) && !defined(PLATFORM_CLANG)
```

Comment:
We have set up our system so that PLATFORM_GCC implies "GCC or compatible" which is confusing, but apps (and possibly external people) depend on it. This variable explicitly tells us if the compiler is GCC. This is the recommend way:
http://nadeausoftware.com/articles/2012/10/c_c_tip_how_detect_compiler_name_and_version_using_compiler_predefined_macros